src/main/java/com/kodilla/ecommercee/controller/GroupController.java

```diff
@@ -0,0 +1,42 @@
+package com.kodilla.ecommercee.controller;
+
+import com.kodilla.ecommercee.domain.GroupDto;
+import com.kodilla.ecommercee.mapper.GroupMapper;
+import com.kodilla.ecommercee.service.DbService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@RestController
+@RequestMapping("v1/group")
+@CrossOrigin("*")
+public class GroupController {
+
+    @Autowired
+    private DbService service;
+
+    @Autowired
+    private GroupMapper groupMapper;
+
+    @RequestMapping (value = "v1/group/getGroupList", method = RequestMethod.GET)
+    public List<GroupDto> getGroupList () {
+        return groupMapper.mapToGroupDtoList(service.getAllGroups());
+    }
+    @GetMapping
+    public GroupDto getGroup (Long groupId){
+        return groupMapper.mapToGroupDto(service.getGroupById(groupId));
+    }
+    @DeleteMapping
+    public void deleteGroup (Long groupId){
+        service.deleteById(groupId);
+    }
+    @PostMapping
+    public void createGroup (@RequestBody GroupDto groupDto) {
+        service.saveTask(groupMapper.mapToGroup(groupDto));
+    }
+    @PutMapping
+    public GroupDto updateTask (@RequestBody GroupDto groupDto){
+        return groupMapper.mapToGroupDto(service.saveTask(groupMapper.mapToGroup(groupDto)));
```

Comment:
trochę długa ta linijka i ciężko się to czyta... nie uważasz, że lepiej by było to rozbić na 3 linijki ?

Reply:
W sumie racja. Czyli chodzi Ci tylko aby to rozbić np. tak:
return groupMapper.mapToGroupDto(
                 service.saveTask( 
                    groupMapper.mapToGroup(groupDto)
     )
);
?

Chyba, że lepiej utworzyć zmienne i je po przekazywać dalej:
@PutMapping
    public GroupDto updateTask (@RequestBody GroupDto groupDto){
    var groupMapper = groupMapper.mapToGroup(groupDto);
    var saveTask = service.saveTask(groupMapper);
        return groupMapper.mapToGroupDto(saveTask);

?

Reply:
Var w java8 nie zadziała. Ale w kilku linijkach jest lepiej w mojej ocenie